### ISSUE TITLE
Unhelpful error message for invalid interaction responses (e.g. using both ephemeral and embed)

### ISSUE THREAD

**Issue by TheJaffaMeme** [Breq16/flask-discord-interactions#5]
Wondering if you could shed some light on this, when I specify that a command should be ephemeral by adding `ephemeral=True` I get a generic "This interaction failed" from discord and there are no errors spit out from python. Wondering if it still works for you with your example?

**Comment by TheJaffaMeme** [comment]
Here's my code resulting in "This interaction failed" when used with `ephemeral=True`

```
## Discord Slash Command Required Imports
from main import discord
from main import logger
from flask_discord_interactions import (Response,
                                        CommandOptionType)

@discord.command()
def info(ctx):
    "Bot Info"
    logger.info(f"/info ran by user '{ctx.author.id}' in guild '{ctx.guild_id}' with parameter(s) ''")
    return Response(ephemeral=True, embed={
            "author": {
                "name": "SlashBot",
                "icon_url": "https://cdn.discordapp.com/app-icons/818189563536343081/6a04d974b41fa3ce18e87ef71dd4b5b9.png?size=256"
            },
            "fields": [
                {
                    "name": "What?",
                    "value": "SlashBot is a bot that uses the new discord slash commands stuffs that aims to be a sort of swiss army knife bot"
                },
                {
                    "name": "Why?!",
                    "value": "NotSoBot was wank and I liked the image commands to terror people"
                },
                {
                    "name": "Where?!?!?!?!",
                    "value": "SlashBot runs off Google Cloud Run"
                },
                {
                    "name": "WHOMST?!?!?!?!",
                    "value": "https://github.com/TheJaffaMeme"
                },
            ],
            })

```

**Comment by TheJaffaMeme** [comment]
It may look odd and that's because I use seperate py files for each command with my bot, it works fine without `ephemeral=True`

**Comment by Breq16** [comment]
So, to be clear, there are no errors in Python, the request shows up in the Flask logs as a 200, but Discord shows an error message?

I noticed this exact behavior sometimes when I was testing out the new response types. I couldn't really figure out what was causing it.

I think what's happening is: Flask and my library receives the interaction from Discord, and returns a response fine (hence, there are no errors in Python). However, this response is somehow invalid, so when Discord receives it, it shows an error.

I thought I had solved this issue, but it looks like I need to do a bit more troubleshooting. I think Discord changed the way they parse responses, since things that used to work have stopped working for me.

Thanks for the detailed report! I'll try to look into this tonight if I have time, but I really can't promise anything--its hard to narrow down the issue because Discord doesn't give any information as to *why* the response is bad.

In the meantime, could you try a response that just includes content and doesn't include an embed? I don't think I've tested embeds in ephemeral responses, so there might be some weird behavior here (either on Discord's end or with my library) that I'm not aware of.

**Comment by TheJaffaMeme** [comment]
Yeah, there are no errors in python and request is logged as 200. Sure enough doing it without an embed makes the ephemeral response work correctly, very odd.

Here's the code that allowed for the ephemeral response to go through if you'd like to compare:

```
## Discord Slash Command Required Imports
from main import discord
from main import logger
from flask_discord_interactions import (Response,
                                        CommandOptionType)

@discord.command()
def test(ctx):
    "Bot Testing"
    logger.info(f"/test ran by user '{ctx.author.id}' in guild '{ctx.guild_id}' with parameter(s) ''")
    return Response("test", ephemeral=True)
```

**Comment by TheJaffaMeme** [comment]
@Breq16  Turns out you cant send ephemeral responses that are embeds, it's a discord problem and not the library see https://github.com/discord/discord-api-docs/issues/2318

**Comment by Breq16** [comment]
Oh, that makes sense. It's a shame Discord hasn't documented their API better. Thanks for your help with narrowing this down!

I'm going to reopen this issue, since I think the library should probably raise its own error here. It's not fun to debug problems like this, so now that we've figured out the root issue, I might as well add some code that verifies the response and shows a more helpful error message in the Python console. (No further action is required on your part--thanks for your report, and I'll try to work on this enhancement sometime soon!)

**Comment by Breq16** [comment]
Fixed in d3e19b8e.
